ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGen/policyBuilder.go

```diff
@@ -41,8 +41,15 @@ func (pbuilder *PolicyBuilder) Build(policyGenTemp utils.PolicyGenTemplate) (map
 			path := policyGenTemp.Metadata.Name + "/" + policyGenTemp.Metadata.Name + "-" + sFile.PolicyName
 			if sFile.PolicyName == "" {
 				// Generate CR without append to policy
-				path := utils.CustomResource + "/" + path
-				policies[path] = resources
+				for _, resource := range resources {
```

Comment:
I don't think there is a need for loop. if it is multi yaml structure just use the first one. wt do u think ?

Reply:
The reason for this loop was to create a single file per resource instead of one file for all the resources within the template, this section deals with the directly applied CRs, and as discussed we wanted to keep them separate.

Reply:
I think we need to separate this CRs into individual files as they will be applied independently on the cluster using `oc apply <fileName>.yaml` . Based on our discussion in the working session we want to keep this implementation so as to work through the deployment on a file by file basis? @imiller0 what do you think?

Reply:
I have removed the loop from the logic replaced with a size check and processing it as an array of size 1